timestamp-impl/src/main/java/com/palantir/timestamp/LastReturnedTimestamp.java

```diff
@@ -23,10 +26,13 @@ public LastReturnedTimestamp(long timestamp) {
     }
 
     public synchronized void increaseToAtLeast(long newTimestamp) {
-        this.timestamp = Math.max(this.timestamp, newTimestamp);
+        long current = this.timestamp;
+        if (newTimestamp > current) {
+            this.timestamp = newTimestamp;
```

Comment:
OK. I don't suppose there are significant gains to do away with `current` altogether?